### ISSUE TITLE
Something in tmp/cache/assets/sprockets/v3.0 causes silent failure of draft up

### ISSUE THREAD

**Issue by kingdonb** [Azure/draft#337]
There are a large number of files in `tmp/cache/` as it tends to get when you do local rails dev, they are gumming up the works and output looks like this:

```
$ draft up
Draft Up Started: 'zinc-sparrow'
zinc-sparrow: Build ID:
```

Clearing everything out of `tmp/cache/assets/sprockets/v3.0/` appears to allow the build to proceed.  Maybe something about the number of files in there, or the length of their names and path combined... this was not so easy to discover, there is no output at all from `dind` or `draftd` in kubectl logs.

So I figured it must be failing before it reaches docker build step... possible that something goes wrong when tarring up the build.tar.gz to make this failure.  I am keeping a copy of the sprockets dir in case it is uncertain what in this directory causes the failure (number of files, or length of filename, or whatever else it could be...)

**Comment by kingdonb** [comment]
For a quick fix, adding `tmp/cache/` to `.dockerignore` seems to be the trick I needed to know.  (Learned this from [build/build.go](https://github.com/Azure/draft/blob/7615dc08bf44510516853a5ce39464f96bb5a397/pkg/build/build.go)) – just adding the path to `.draftignore` does not appear to help, and I'm not sure why.

**Comment by bacongobbler** [comment]
Perhaps we should rename .draftignore to .watchignore, or just read from .gitignore instead. .draftignore is only used with `--watch`, and it does seem to be causing a bit of confusion.

Adding `tmp/` to .gitignore should cause this to not fail, but I believe this should be resolved with #226.

**Comment by kingdonb** [comment]
I would prefer not to use .gitignore, because I have credentials in .env.local that I'm expecting to get uploaded to the image, but I do not want them to wind up in the git repository.

I like `.watchignore`, that communicates better what `.draftignore` is supposed to do.  The real problem for me was that `.dockerignore` was difficult to find.  I had to read the source code.  (The real problem is actually that it failed silently, but I think you're right that will be addressed by #226.)

Rather than changing the API, maybe better to just have it better documented which ignore files are relevant and what stages they pertain to.  The file `.dockerignore` seems like the right thing to use, is already a standard in the Docker documentation, and is documented there, but as Draft also takes some steps to honor the `.dockerignore` it might be worth a mention somewhere in the final Draft documentation.
